src/components/utils.js

```diff
@@ -22,6 +33,25 @@ export function fetchArticles(filter = null) {
   })
 }
 
+export function fetchFilteredArticles(topic = null, sort = null) {
+  let Query = null
+  if (topic.topic !== undefined && sort.sort_by !== null ) {
+    Query = `?${Object.keys(topic)}=${Object.values(topic)}&${Object.keys(sort)}=${Object.values(sort)}`
+  }
+  else if(sort.sort_by !== null && topic.topic === undefined) {
+    Query = `?${Object.keys(sort)}=${Object.values(sort)}`
+  }
+  else if (sort.sort_by === null && topic.topic !== undefined) {
+    Query = `?${Object.keys(topic)}=${Object.values(topic)}`
```

Comment:
Please look into the use of axios 'params' in the Notes and try to refactor this.